### PR TITLE
Update vscode engine and test runners

### PR DESCRIPTION
**Pull request by dylanbeattie**
This updates the vscode engine to 1.87.0, the TypeScript compiler target to ES2022, and replaces the old Mocha test runner with the new VS Code test runner.

(Presentation Buddy is *awesome* and I'm working on a new feature for it, which needed a working test runner to figure out some parsing things. I'm happy to build off my own fork if this PR is going to break your publishing flow or anything but figured I'd open the PR and see what you thought!)

### CODE REVIEW

**Comment by mauricedb** [comment]
Cool, I really appreciate the pr. I am a little bit busy at the moment but will review and get back to you as soon as possible.

**Comment by dylanbeattie** [comment]
So the actual feature I propose to add is implemented here:

https://github.com/dylanbeattie/presentation-buddy/pull/1

(I haven't opened a PR for this one yet because until the tooling PR is merged, the diffs between my branch and the upstream master are a bit of a mess)

Notes on what it adds (and how it works) are in the README - take a look, I'd love to know whether you think this is a useful addition to Presentation Buddy or if you'd prefer I spin it off as a new extension.

Thanks 😊

File: src/test/index.ts
Comment (mauricedb) [inline src/test/index.ts:2]: 🧹 Looks like this file can be deleted

**Review by mauricedb — CHANGES_REQUESTED** [review]
I am fine with merging this PR except for one thing. I would prefer to use NPM as the package manager instead or Yarn V1.

**Comment by mauricedb** [comment]
Hi Dylan,

I was just looking at the *typeChunksFromFile* instruction you are
proposing and I like that way it works. Looks like that will make it much
easier to use with a large amount of input where you want additional
pauses, something I do quite often.

I would probably not have added the special case for the *waitAfterNewLine*
as it looks like it could just as easily be included in the
*waitAfterTyping* array but I don't have an issue with it either.

I would be delighted to include this PR.

Thank,
Maurice


Maurice de Beijer
ABL – The Problem Solver
Microsoft MVP
www.TheProblemSolver.nl
www.twitter.com/mauricedb

**Comment by dylanbeattie** [comment]
Hey Maurice,

There are two reasons I plugged in WaitAfterNewLine as a special case. First, having used this for a week or so I didn't actually find any scenario or language where I *didn't* want that option, so having it enabled by default seemed like a good idea.

Second: the VS Code configuration editor for string arrays can't cope with literal newlines - if you put a `\n` in the JSON config, it shows as an empty string in the config editor, and if you enter a `\n` in the config editor UI it's persisted as `\\n` in the JSON.

I agree it would make more sense as just another entry in the array, but after playing around with it I decided the tooling made this more confusing than having an extra config setting - let me know what you think.

File: src/test/index.ts
Comment (dylanbeattie) [reply src/test/index.ts:2]: Sure, I can update the PR over the weekend to use npm. The existing code used a weird mix of yarn and npm and I wasn't sure if this was to support a particular build pipieline or something and didn't want to change it unnecessarily, but npm will work just fine - I'll ping you when it's updated.

Comment (mauricedb) [reply src/test/index.ts:2]: I was not aware of a mix between Yarn and NPM. That, must have been by mistake, my bad.

**Comment by mauricedb** [comment]
> Hey Maurice,
> 
> There are two reasons I plugged in WaitAfterNewLine as a special case. First, having used this for a week or so I didn't actually find any scenario or language where I _didn't_ want that option, so having it enabled by default seemed like a good idea.
> 
> Second: the VS Code configuration editor for string arrays can't cope with literal newlines - if you put a `\n` in the JSON config, it shows as an empty string in the config editor, and if you enter a `\n` in the config editor UI it's persisted as `\\n` in the JSON.
> 
> I agree it would make more sense as just another entry in the array, but after playing around with it I decided the tooling made this more confusing than having an extra config setting - let me know what you think.

👍 Right makes total sense. 

Just leave it as is then and I will be happy to merge this when you create the second PR.

File: src/test/index.ts
Comment (dylanbeattie) [reply src/test/index.ts]: In that case, my best guess is the (Yeoman?) template you originally used to create the extension was designed to work with yarn but worked just fine with npm as long as nobody tried to run the yarn-specific commands in the package.json scripts section.

It's all on npm now; merge whenever you're ready and I'll create the new PR with the typeChunksFromFile feature support. 👍🏼

**Review by mauricedb — APPROVED** [review]

File: src/test/index.ts
Comment (mauricedb) [reply src/test/index.ts]: Yes I used Yeoman when I first created this project to get started. And as this was just a simple utility to help with presentations I never used any proper engineering principles. Just sort of hacked it together.

**Pull request merged**
